src/Nethermind/Nethermind.BeaconNode.OApi/BeaconNodeOApiAdapter.cs

```diff
@@ -75,42 +75,62 @@ public async Task<BeaconBlock> BlockAsync(ulong slot, byte[] randao_reveal)
         /// <summary>Get validator duties for the requested validators.</summary>
         /// <param name="validator_pubkeys">An array of hex-encoded BLS public keys</param>
         /// <returns>Success response</returns>
-        public Task<ICollection<ValidatorDuty>> DutiesAsync(System.Collections.Generic.IEnumerable<byte[]> validator_pubkeys, int? epoch)
+        public async Task<ICollection<ValidatorDuty>> DutiesAsync(System.Collections.Generic.IEnumerable<byte[]> validator_pubkeys, int? epoch)
         {
-            throw new NotImplementedException();
+            IEnumerable<BlsPublicKey> publicKeys = validator_pubkeys.Select(x => new BlsPublicKey(x));
+            Epoch targetEpoch = epoch.HasValue ? new Epoch((ulong)epoch) : Epoch.None;
+            IList<BeaconNode.ValidatorDuty> duties = await _beaconNode.ValidatorDutiesAsync(publicKeys, targetEpoch);
+            List<ValidatorDuty> result = duties.Select(x =>
+                {
+                    ValidatorDuty validatorDuty = new ValidatorDuty();
+                    validatorDuty.Validator_pubkey = x.ValidatorPublicKey.Bytes;
+                    validatorDuty.Attestation_slot = (int)x.AttestationSlot;
+                    validatorDuty.Attestation_shard = (int)(ulong)x.AttestationShard;
+                    validatorDuty.Block_proposal_slot = x.BlockProposalSlot == Slot.None ? null : (int?)x.BlockProposalSlot;
+                    return validatorDuty;
+                })
+                .ToList();
+            return result;
         }
 
         /// <summary>Get fork information from running beacon node.</summary>
         /// <returns>Request successful</returns>
-        public Task<Response2> ForkAsync()
+        public async Task<Response2> ForkAsync()
         {
-            throw new NotImplementedException();
+            Core2.Containers.Fork fork = await _beaconNode.GetNodeForkAsync();
+            Response2 response2 = new Response2();
+            // TODO: Not sure what chain ID should be.
+            response2.Chain_id = 0;
+            response2.Fork = new Fork();
+            response2.Fork.Epoch = fork.Epoch;
+            response2.Fork.Current_version = fork.CurrentVersion.AsSpan().ToArray();
+            response2.Fork.Previous_version = fork.PreviousVersion.AsSpan().ToArray();
+            return response2;
         }
 
         /// <summary>Poll to see if the the beacon node is syncing.</summary>
         /// <returns>Request successful</returns>
-        public Task<Response> SyncingAsync()
+        public async Task<Response> SyncingAsync()
         {
-            throw new NotImplementedException();
+            Response response = new Response();
+            response.Is_syncing = await _beaconNode.GetIsSyncingAsync();
+            response.Sync_status = new SyncingStatus();
```

Comment:
I wonder if it is possible to configure the code generator to generate SyncStatus instead of Sync_status

Reply:
Not sure. With older XML stuff you could point to an existing class that implemented the schema, so I used to run the generator, then go through and change to .NET naming standards with the attribute with the API name.

I'll see if something similar exists in JSON. About the limit of what I've done in the past is turn on camel case, i.e. "FooBar" -> "fooBar".